### DISCUSSION TITLE
RaspberryMatic-3.67.9.20221229-813312-rpi3 - WebUI fehler

### DISCUSSION THREAD

**Discussion by ghost** [OpenCCU/OpenCCU#2130] [🇩🇪 Fragen&Antworten] [answered]
![Animation1](https://user-images.githubusercontent.com/81189856/209936550-5e666e4f-aa76-43b2-8b1d-3af231bb9b9a.gif)
 keine weiteren Favoriten auswählbar (nur der erste Reiter) - im chrome oder firefox

**Comment by jens-maus** [comment]
Und was steckt eigentlich hinter diesem Favoriten? Und geht es noch mit der aktuellen Release Version oder mit einer Vorversion?

**Comment by ghost** [comment]
Der Fehler betrifft nur die letzte Version Nightly RaspberryMatic-3.67.9.20221229-813312-rpi3 , 
und so sollte es richtig aussehen (RaspberryMatic-3.67.9.20221221-c4087b-rpi3 ) :

![Animation1](https://user-images.githubusercontent.com/81189856/209941012-d3dc07ff-bafc-4f65-a133-9b28a21df5dc.gif)


In dieser Version RaspberryMatic-3.67.9.20221221-c4087b können Sie einzelne Registerkarten öffnen , In dieser Version RaspberryMatic-3.67.9.20221229-813312 konnten Sie nur die erste Seite der Favoriten anzeigen und nicht den Rest

**Reply by ghost** [reply]
Bitte schalten Sie Animationen ein, wenn sie nicht automatisch eingeschaltet werden

**Comment by Baxxy13** [comment]
Also ich kann das gezeigte Problem mit dem heutigen Nightly reproduzieren.
In meinen 3 Favoriten sind nur SysVars und ein paar CUxD-Geräte.

Egal welchen Favoriten ich anwähle (auf der Startseite wie oben zu sehen), es wird immer nur der "erste" Favorit aufgerufen. 
Also der Favorit der in "> Einstellungen >Favoriten" die Position 1 hat. 

Es gibt aber noch ein weiters Problem.
Klicke ich in "> Status und Bedienung > Favoriten" auf einen meiner Favoriten so wird dieser nicht geöffnet sondern es kommt der Text:
`Kann Favoriten nicht öffnen!`
![GIF 29 12 2022 18-03-12](https://user-images.githubusercontent.com/54898249/209985582-0bca0416-9799-4b87-84fe-eb0859c3e78c.gif)

Fehlermeldungen in der Browserkonsole gibt es keine.

**Comment by ghost** [comment]
und in dieser Nightly RaspberryMatic-3.67.9.20221221-c4087b-rpi3 war noch alles in Ordnung

**Reply by Baxxy13** [reply]
Noch präziser: `3.67.9.20221228-a53bf0` war diesbezüglich auch noch in Ordnung.
Damit bleiben bloß noch 2 commits die als Verursacher infrage kommen.

**Comment by Baxxy13** [comment, marked as answer]
Mit ein bisschen Hilfe konnte das Problem inzwischen gefunden (und manuell behoben) werden.
Beim Auswählen des Favoriten wurde die `fid` nicht mehr mit übergeben.

Falscher Aufruf: `http://192.168.1.45/pages/tabs/startpage.htm?sid=@DbDmvAxSUp@`
Korrekter Aufruf: `http://192.168.1.45/pages/tabs/startpage.htm?sid=@DbDmvAxSUp@&fid=18963`

Danach war es einfach und recht offensichtlich... 

#https://github.com/jens-maus/RaspberryMatic/commit/41bca8a9901e35c89736b4df43e0da316ab75ce3#diff-c56b9320e8f7dd7acbb80e913048cdfcfce43566f1dc13dfdd2a952994dfefeaR75

Manuelles Einfügen von `if (argsForUrl){ url += argsForUrl; }` an der richtigen Stelle in der webui.js behebt die beiden Probleme.
